### DISCUSSION TITLE
Transmuting to static mut reference: Is it correct?

### DISCUSSION THREAD

**Discussion by Nugine** [web-infra-dev/rspack#2122] [Q&A] [answered]
I cannot find any safety comments here.

https://github.com/web-infra-dev/rspack/blob/2f7ffb57066b536d187e9d555c1ac4813fed7081/crates/node_binding/src/plugins/mod.rs#L39-L59

**Comment by h-a-n-a** [comment, marked as answer]
This is considered to be safe under the `compilation` call from the JavaScript side. But It is not a safe if you are trying to cache the compilation and reuse it after any rebuilds, which would cause some potential segmentation faults.

**Reply by h-a-n-a** [reply]
Feel free to come up with a PR to describe this SAFETY contract. We would be much appreciated :)
